src/transformState.js

```diff
@@ -5,7 +5,26 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'addProperties') {
+      for (const prop in actions[i].extraData) {
+        state[prop] = actions[i].extraData[prop];
+      }
+    }
+
+    if (actions[i].type === 'removeProperties') {
+      for (const word in actions[i].keysToRemove) {
+        delete state[actions[i].keysToRemove[word]];
```

Comment:
and then you'd only need
```suggestion
        delete state[word];
```